oidcclient/jwk.go

```diff
@@ -51,12 +50,16 @@ func NewOIDCTenant(httpClient *http.Client, targetIss *url.URL) (*OIDCTenant, er
 }
 
 // GetJWKs returns the validation keys either cached or updated ones
-func (ks *OIDCTenant) GetJWKs() (jwk.Set, error) {
-	if time.Now().Before(ks.jwksExpiry) {
-		return ks.jwks, nil
-	}
+func (ks *OIDCTenant) GetJWKs(zoneID string) (jwk.Set, error) {
+	isZoneAccepted, ok := ks.acceptedZoneIds[zoneID]
 
-	updatedKeys, err, _ := ks.singleFlight.Do("updateKeys", ks.updateKeys)
+	if time.Now().Before(ks.jwksExpiry) && ok {
+		if isZoneAccepted {
+			return ks.jwks, nil
+		}
+		return nil, fmt.Errorf("severe security issue: zone_uuid %v is still not accepted", zoneID)
```

Comment:
Why "still" not accepted? I think an error message in the style of the other error messages suites better. e.g.
token validation failed: zone_uuid %s is not accepted by the identity service tenant

Btw: Should we maybe add a JWK update if the zone_uuid is not yet in the acceptedZoneIds map? Otherwise a new onboarded (subscribed) tenant with a new zone_uuid would have to wait 15mins in the worst case until he can access the application.

Reply:
we need to delete the zones regularly from cache as zones can be also deleted.... Right now they get deleted - like the JWKS - every 15 Minutes.

Reply:
i've removed "still from error message but kept "severe security issue" as caller retries with "deleted/wrong" zoneId which should be written as security event to audit log. Do you have a better way to do that?

Reply:
Is it a security issue? We handle the scenario and the token is rejected like any other invalid token. So nothing the application can or has to do about it. 
It's basically the same as if you call the application multiple times with an invalid signature in the token.